sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java

```diff
@@ -32,12 +32,21 @@
   private final String[] qualifiers;
   private final String language;
 
-  public ComponentQuery(@Nullable String nameOrKeyQuery, String language, String... qualifiers) {
-    this.language = language;
-    checkArgument(qualifiers.length > 0, "At least one qualifier must be provided");
-
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Dev Cockpit doesn't use it anymore
```

Comment:
was about to say there should be a `@deprecated` javadoc tag, but @SonarTech was faster